addons/mail/models/discuss/discuss_channel_member.py

```diff
@@ -163,12 +180,36 @@ def _compute_is_pinned(self):
                 )
             )
 
+    @api.depends("channel_ownership")
+    def _compute_channel_ownership_in_view(self):
+        for member in self:
+            member.channel_ownership_in_view = member.channel_ownership
+
+    def _inverse_channel_ownership_in_view(self):
+        current_user_channel_member = self.env["discuss.channel.member"].search([
+            ("channel_id", "=", self.channel_id.id),
+            ("partner_id", "=", self.env.user.partner_id.id),
+        ])
+        if not current_user_channel_member:
+            raise AccessError(_("You are not a member of this channel."))
```

Comment:
The check should be done per member/channel, as there is a loop of potentially different members.

Reply:
You need to have a mapping of current user member per channel.